syntax = "proto3";

enum RequestCode
{
    UserLogin = 0;
    Register = 1;
    AesFenfa = 2;
    AutoRoom = 3;
    ManualRoom = 4;
    SearchRoom = 5;
};

enum RespondCode
{
    LoginOk = 0;
    RegisterOk = 1;
    RsaFenFa = 2;
    AesVerifOK = 3;
    JoinRoomOK = 4;
    StartGame = 5;
    SearchRoomOK = 6;
    DealCards = 7;
    Failed = 8;
};

message Information
{
    string userName = 1;
    string roomName = 2;
    string data1 = 3;
    string data2 = 4;
    string data3 = 5;
    RequestCode reqcode = 6;
    RespondCode rescode = 7;
};
